File: python/examples/assess_company.py
"""Assess one company and print a triage decision.

Run with:

    export KYCCENTRAL_API_KEY="your_key"
    python examples/assess_company.py 00445790
"""

from __future__ import annotations

import sys

from kyccentral import KYCCentral, KYCCentralError, RiskLevel


def main(company_number: str) -> int:
    with KYCCentral() as client:
        try:
            assessment = client.kyc.assess(company_number)
        except KYCCentralError as exc:
            print(f"Assessment failed: {exc}", file=sys.stderr)
            return 1

    print(f"{assessment.company_name} ({assessment.company_number})")
    print(f"Overall risk: {assessment.risk_level.value}")
    print(f"Registry data as of: {assessment.data_fetched_at}")
    print()

    if assessment.flags:
        print(f"{len(assessment.flags)} flag(s):")
        # Most severe first, so the blockers are at the top of the output.
        for flag in sorted(assessment.flags, key=lambda f: f.severity.rank, reverse=True):
            print(f"  [{flag.severity.value:<8}] {flag.code}: {flag.description}")
    else:
        print("No flags raised.")

    if assessment.is_partial:
        print()
        print("WARNING — this assessment is incomplete, so an absent flag is not a clear result.")
        if assessment.timed_out_services:
            print(f"  Timed out: {', '.join(assessment.timed_out_services)}")
        if assessment.failed_rules:
            print(f"  Rules that errored: {', '.join(assessment.failed_rules)}")
        if assessment.pending_extractions:
            print(f"  Still extracting: {', '.join(assessment.pending_extractions)}")

    print()
    if assessment.critical_flags:
        print("Decision: BLOCK — critical findings require escalation.")
        return 2
    if assessment.is_partial or assessment.risk_level is not RiskLevel.LOW:
        print("Decision: REFER for manual review.")
        return 0
    print("Decision: CLEAR.")
    return 0


if __name__ == "__main__":
    if len(sys.argv) != 2:
        print(f"usage: {sys.argv[0]} <company_number>", file=sys.stderr)
        raise SystemExit(64)
    raise SystemExit(main(sys.argv[1]))

File: javascript/examples/assess-company.ts
/**
 * Assess one company and print a triage decision.
 *
 * Run with:
 *
 *     export KYCCENTRAL_API_KEY="your_key"
 *     npx tsx examples/assess-company.ts 00445790
 */

import {
  KYCCentral,
  KYCCentralError,
  RISK_LEVEL_RANK,
  isPartial,
  type RiskFlag,
} from '../src/index.js';

async function main(companyNumber: string): Promise<number> {
  const client = new KYCCentral();

  let assessment;
  try {
    assessment = await client.kyc.assess(companyNumber);
  } catch (error) {
    if (error instanceof KYCCentralError) {
      console.error(`Assessment failed: ${error.message}`);
      return 1;
    }
    throw error;
  }

  console.log(`${assessment.companyName} (${assessment.companyNumber})`);
  console.log(`Overall risk: ${assessment.riskLevel}`);
  console.log(`Registry data as of: ${assessment.dataFetchedAt}`);
  console.log();

  if (assessment.flags.length > 0) {
    console.log(`${assessment.flags.length} flag(s):`);
    // Most severe first, so the blockers are at the top of the output.
    const bySeverity = [...assessment.flags].sort(
      (a: RiskFlag, b: RiskFlag) => RISK_LEVEL_RANK[b.severity] - RISK_LEVEL_RANK[a.severity],
    );
    for (const flag of bySeverity) {
      console.log(`  [${flag.severity.padEnd(8)}] ${flag.code}: ${flag.description}`);
    }
  } else {
    console.log('No flags raised.');
  }

  if (isPartial(assessment)) {
    console.log();
    console.log(
      'WARNING — this assessment is incomplete, so an absent flag is not a clear result.',
    );
    if (assessment.timedOutServices.length) {
      console.log(`  Timed out: ${assessment.timedOutServices.join(', ')}`);
    }
    if (assessment.failedRules.length) {
      console.log(`  Rules that errored: ${assessment.failedRules.join(', ')}`);
    }
    if (assessment.pendingExtractions.length) {
      console.log(`  Still extracting: ${assessment.pendingExtractions.join(', ')}`);
    }
  }

  console.log();
  const critical = assessment.flags.filter((flag) => flag.severity === 'critical');
  if (critical.length > 0) {
    console.log('Decision: BLOCK — critical findings require escalation.');
    return 2;
  }
  if (isPartial(assessment) || assessment.riskLevel !== 'low') {
    console.log('Decision: REFER for manual review.');
    return 0;
  }
  console.log('Decision: CLEAR.');
  return 0;
}

const companyNumber = process.argv[2];
if (!companyNumber) {
  console.error('usage: assess-company.ts <company_number>');
  process.exit(64);
}

main(companyNumber).then(
  (code) => process.exit(code),
  (error) => {
    console.error(error);
    process.exit(1);
  },
);
